pkg/state/appender.go

```diff
@@ -717,6 +717,13 @@ func (a *txAppender) countExchangeScriptsRuns(scriptsRuns uint64) (uint64, error
 	return scriptsRuns, nil
 }
 
+func checkAttachment(o proto.Order) bool {
+	if ov4, ok := o.(*proto.OrderV4); ok && ov4.Attachment.Size() == 0 {
+		return true
+	}
+	return false
+}
```

Comment:
It's not correct because we also have `proto.EthereumOrderV4` type.
The best option here, on my mind, is:
```suggestion
func checkAttachmentLengthIsZero(o proto.Order) bool {
	return o.GetAttachment().Size() != 0
}
```